gapic-common/lib/gapic/common/polling_harness.rb

```diff
@@ -0,0 +1,67 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Gapic
+  module Common
+    ##
+    # Provides a generic mechanism for periodic polling an operation.
+    #
+    # Polling intervals are calculated from the provided retry policy.
+    #
+    # Supports exponential backoff via `multiplier`, and automatically
+    # retries gRPC errors listed in `retry_codes`.
+    #
+    class PollingHarness
+      # @return [Gapic::Common::RetryPolicy] The retry policy associated with this instance.
+      attr_reader :retry_policy
+
+      ##
+      # Create new Gapic::Common::PollingHarness instance.
+      #
+      # @param retry_policy [Gapic::Common::RetryPolicy]
+      #
+      def initialize retry_policy:
+        @retry_policy = retry_policy
+      end
+
+      ##
+      # Perform polling with exponential backoff.
+      #
+      # The provided block MUST evaluate to `nil` in cases where the operation
+      # should continue. Otherwise, the evaluated result will be returned.
+      #
+      # Errors are raised if not listed in `retry_codes`, and re-attempted otherwise.
+      #
+      # The retry policy's `timeout` value is _NOT_ taken into consideration. The block
+      # will be re-attempted indefinitely as long as the retry conditions above
+      # passes.
+      #
+      # @yieldreturn [Object, nil] Custom logic to be retriable.
+      def wait
+        unless block_given?
+          raise ArgumentError, "No callback provided to wait method."
+        end
+        loop do
+          begin
+            response = yield
+            return response unless response.nil?
+          rescue ::GRPC::BadStatus => e # Currently retry_error only account for this type.
```

Comment:
We don't have a guarantee that grpc is loaded (since we might be using an HTTP/REST GAPIC client) at which point this check itself could raise a constant-not-defined exception.

To fix this, just rescue `StandardError` instead. (i.e. `rescue => e`. You might need to disable a rubocop check for that line.) This should not affect the functionality because `RetryPolicy#retry_error?` already checks for the `GRPC::BadStatus` exception type.

Reply:
Updated it to `StandardError`. Thanks for the review btw, please feel free to add anything else!